eclipse-cdt-cloud/vscode-memory-inspector
# Proposal: Switch bundling to ESBuild

#### What it does


This PR switches bundling from Webpack to ESBuild.

Pros:
 1. Much, much faster: On my machine, `yarn build` runs in 0.41s vs. 24 seconds with Webpack.
 3. Smaller bundle: `dist` (with source maps) about 12% smaller.
 4. More transparent bundle: everything is in `dist`; nothing additional packaged from `node_modules`.
 5. Fewer dependencies.

Cons:
 1. Type checking is not part of build: added separate `types` script to check with `tsc`.
 2. Can struggle with dynamic imports (but so can Webpack). I have run afoul of this problem in the past, but there don't appear to be any dynamic `require` calls in the bundled code in this repo.
 3. Because type checking isn't part of the build, running the `watch` task won't produce error output if you introduce errors in development. If that's an important part of people's workflow, we can run `tsc --watch` in parallel with ESBuild's watch.

#### How to test

1. Local smoke tests - do all styles & icons in webview appear correct?
2. Package and install + smoke tests - ensure that we're not cheating and using local files that wouldn't be packaged.
> Pay special attention to browser plugin use case - I'm less familiar with its behavior, as we use primarily the desktop package.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the code of conduct](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/blob/main/CODE_OF_CONDUCT.md)
